admin/tabs/userData/roles/recategorize/RecategorizeDialogModel.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2024 Extremely Heavy Industries Inc.
+ */
+import {RoleModel} from '@xh/hoist/admin/tabs/userData/roles/RoleModel';
+import {HoistModel, XH} from '@xh/hoist/core';
+import {Icon} from '@xh/hoist/icon/Icon';
+import {action, bindable, makeObservable, observable} from '@xh/hoist/mobx';
+import {compact, every, filter, map, uniq} from 'lodash';
+
+export class RecategorizeDialogModel extends HoistModel {
+    _parent: RoleModel;
+
+    @bindable categoryName = null;
+    @observable isOpen = false;
+
+    reCategorizeAction() {
+        return {
+            text: 'Change Category',
+            icon: Icon.folder(),
+            recordsRequired: true,
+            actionFn: () => this.open(),
+            displayFn: ({selectedRecords}) => {
+                return {
+                    hidden: this._parent.readonly,
+                    disabled: every(selectedRecords, it => it.data?.isGroupRow)
+                };
+            }
+        };
+    }
+
+    get options() {
+        return compact(uniq(this._parent.allRoles.map(it => it.category))).sort();
+    }
+
+    constructor(parent) {
+        super();
+        makeObservable(this);
+        this._parent = parent;
+    }
+
+    async saveAsync(clear: boolean = false) {
+        if (this._parent.readonly) return;
+        const roleSpec = filter(
+            this._parent.gridModel.selectedRecords.map(it => it.data),
```

Comment:
This makes me just a little bit nervous, reading the selected records here, right before saving. I don't know of any reason why they would change underneath the dialog, but it feels like a somewhat unsafe pattern.

Might be a better pattern to have `open` take the role names to be changed and install those on this model as some temp state (to be cleared on close).  Then we've captured the user's intention from the start. Could print the count in the dialog header - "Change Category (8 roles)"

To be clear - practically speaking I don't think there's any risk, but we should strive to make this Hoist admin code as clear and robust as possible, both to ensure it's reliable and as an example of strong patterns for anyone reading it.